### PR TITLE
Java + spring upgrade

### PR DESCRIPTION
**Pull request by melmi-ch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,42 +6,46 @@
     <parent>
         <groupId>uk.gov.companieshouse</groupId>
         <artifactId>companies-house-parent</artifactId>
-        <version>1.3.0</version>
+        <version>2.1.6</version>
```
Comment (foman-ch) [inline pom.xml:9]: there's a 2.1.9 now

File: pom.xml [hunk 2]
```diff
@@ -6,42 +6,46 @@
...
+        <maven-surefire-plugin.version>3.5.2</maven-surefire-plugin.version>
+        <spring-boot-maven-plugin.version>3.3.5</spring-boot-maven-plugin.version>
+        <commons-io.version>2.17.0</commons-io.version>
+        <!-- CH Dependencies-->
         <start-class>uk.gov.companieshouse.documentsigningapi.DocumentSigningApiApplication</start-class>
-
-        <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
-
+        <api-security-java.version>2.0.8</api-security-java.version>
+        <structured-logging.version>3.0.9</structured-logging.version>
```
Comment (foman-ch) [inline pom.xml:32]: there's a 3.0.20

File: pom.xml [hunk 3]
```diff
@@ -223,6 +254,15 @@
...
                 <version>${jib-maven-plugin.version}</version>
                 <configuration>
+                    <container>
+                        <expandClasspathDependencies>true</expandClasspathDependencies>
+                    </container>
+                    <from>
+                        <image>416670754337.dkr.ecr.eu-west-2.amazonaws.com/ci-corretto-build-21:latest</image>
+                    </from>
+                    <to>
+                        <image>169942020521.dkr.ecr.eu-west-2.amazonaws.com/local/document-signing-api:latest</image>
```
Comment (foman-ch) [inline pom.xml:264]: think it's ok to update this now to the new image: 416670754337.dkr.ecr.eu-west-2.amazonaws.com/document-signing-api:latest

File: pom.xml [hunk 4]
```diff
@@ -242,6 +282,14 @@
                     </extraDirectories>
                 </configuration>
             </plugin>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+              <source>21</source>
```
Comment (foman-ch) [inline pom.xml:289]: add as a property :)

**Review by foman-ch — CHANGES_REQUESTED** [review]

File: src/test/resources/application.properties [hunk 5]
```diff
@@ -6,13 +6,11 @@ management.endpoints.web.path-mapping.health=/healthcheck
 uk.gov.companieshouse.documentsigningapi.signpdf=/document-signing/sign-pdf
 
 # Environment variables
-environment.keystore.type=${KEYSTORE_TYPE}
-environment.keystore.path=${KEYSTORE_PATH}
-environment.keystore.password=${KEYSTORE_PASSWORD}
-environment.certificate.alias=${CERTIFICATE_ALIAS}
```
Comment (vshastri-ch) [inline src/test/resources/application.properties:12]: Curious to know why the variables are changed to hardcoded values here?

Comment (melmi-ch) [reply src/test/resources/application.properties:12]: The itest env doesn't pick up the values created after upgrade. Only works when hardcoded in.

**Review by vshastri-ch — APPROVED** [review]

**Pull request merged**